sys/include/entropy_source/adc_noise.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup   sys_entropy_source_adc ADC Noise Entropy Source
+ * @ingroup    sys_entropy_source
+ * @brief      Entropy Source based on LSB of ADC samples
+ *
+ * This module provides entropy from ADC samples. Thereby, only the LSB is considered for
+ * generation of output values. The available resolution as well as internal sampling rates,
+ * specific driver settings and device inaccuracies may lead to different behavior between
+ * different platforms. The configured ADC pin might be unconnected and floating,
+ * it can be exposed as an I/O pin or internally connected to a thermal noise source, an
+ * avalanche diode circuit or the receive path of an antenna. The possibilities are manifold,
+ * thus, configuration and deployment properties of this module need to be thoroughly validated.
+ * It is worth noting that ADC pins are typically vulnerable when physically accessible.
+ * Developers should consider additional tamper detection concepts and enclosures.
+ *
+ * @{
+ * @file
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+
+#ifndef ENTROPY_SOURCE_ADC_NOISE_H
+#define ENTROPY_SOURCE_ADC_NOISE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <inttypes.h>
+#include <stddef.h>
+
+#include "periph/adc.h"
+
+/**
+ * @brief   ADC default configuration for entropy generation.
+ * @{
+ */
+#ifndef ENTROPY_SOURCE_ADC_LINE
+#define ENTROPY_SOURCE_ADC_LINE         ADC_LINE(0)
+#endif
+#ifndef ENTROPY_SOURCE_ADC_RES
+#define ENTROPY_SOURCE_ADC_RES          ADC_RES_10BIT
+#endif
+#ifndef CONFIG_ENTROPY_SOURCE_ADC_HMIN
+#define CONFIG_ENTROPY_SOURCE_ADC_HMIN         (65536)  /**< H_min=1 bit/sample * 2^16
+                                                          *  only as an example!
+                                                          */
+
+#endif
+#ifndef CONFIG_ENTROPY_SOURCE_ADC_HEALTH_TEST
+#define CONFIG_ENTROPY_SOURCE_ADC_HEALTH_TEST  (0)
+#endif
+#ifndef CONFIG_ENTROPY_SOURCE_ADC_COND
+#define CONFIG_ENTROPY_SOURCE_ADC_COND         (0)
+#endif
```

Comment:
```suggestion
#ifdef DOXYGEN
#define CONFIG_ENTROPY_SOURCE_ADC_HEALTH_TEST  (0)
#define CONFIG_ENTROPY_SOURCE_ADC_COND         (0)
#endif 
```

Also consider adding documentation and including them on the compile time configuration group